e2e/tests/test_cases/single_replica_node_down.robot

```diff
@@ -0,0 +1,118 @@
+*** Settings ***
+Documentation    Single replica node down
+...              https://longhorn.github.io/longhorn-tests/manual/pre-release/node-not-ready/node-down/single-replica-node-down/
+
+Test Tags    manual_test_case
+
+Resource    ../keywords/common.resource
+Resource    ../keywords/deployment.resource
+Resource    ../keywords/persistentvolumeclaim.resource
+Resource    ../keywords/recurringjob.resource
+Resource    ../keywords/statefulset.resource
+Resource    ../keywords/volume.resource
+Resource    ../keywords/setting.resource
+Resource    ../keywords/setting.resource
+Resource    ../keywords/workload.resource
+Resource    ../keywords/host.resource
+
+Library    OperatingSystem
+Test Setup    Set test environment
+Test Teardown    Cleanup test resources
+
+*** Variables ***
+${LOOP_COUNT}    1
+${RETRY_COUNT}    300
+${RETRY_INTERVAL}    1
+${SHUTDOWN_WAIT_DURATION}    180
+
+*** Test Cases ***
+Single Replica Node Down Deletion Policy do-nothing With RWO Volume Data Locality disabled
+    When Create persistentvolumeclaim 0 using RWO volume
+    And Create deployment 0 with persistentvolumeclaim 0
+    And Write 100 MB data to file data in deployment 0
+    And Update persistentvolumeclaim 0 volume replica count to ${1}
+    And Delete replica of deployment 0 volume on replica node
+    And Delete replica of deployment 0 volume on volume node
+    And Power off volume node of deployment 0
+    And Sleep   ${SHUTDOWN_WAIT_DURATION}
+    Then Wait for volume of deployment 0 attaching
+
+    When Power on volume node of deployment 0
+    And Wait for deployment 0 pods stable
+    Then Check deployment 0 data in file data is intact
+
+Single Replica Node Down Deletion Policy do-nothing With RWO Volume Data Locality
+    When Create persistentvolumeclaim 0 using RWO volume
+    And Create deployment 0 with persistentvolumeclaim 0
+    And Write 100 MB data to file data in deployment 0
+    And Update persistentvolumeclaim 0 volume replica count to ${1}
+    And Delete replica of deployment 0 volume on all replica node
+    And Power off volume node of deployment 0
+    And Sleep   ${SHUTDOWN_WAIT_DURATION}
+    Then Wait for volume of deployment 0 faulted
+
+    When Power on off nodes
+    And Wait for deployment 0 pods stable
```

Comment:
Will the pod be created on another node or the same node? Will the volume be attached to another node or the same node? 

Does it need to be verified as well?

Reply:
IIRC, we do not have restrict rule for this and the seems the pod lcation decided by scheduler,  therefore check volume/pod OK and data intact maybe enough for now.

Reply:
But the original manual test case has steps to check pod created on another node, right?
```
Power up node or delete the workload pod so that kubernetes will recreate pod on another node.
````

```
Wait for pod deletion & recreation on another node.
```

Reply:
> Will the pod be created on another node or the same node? Will the volume be attached to another node or the same node?
> 
> Does it need to be verified as well?

Added `Wait for ${workload_kind} ${workload_id} pod ${expect_state} on ${replica_locality}` for check